_posts/2020-04-01-Robust Apps for the cloud.md

```diff
@@ -0,0 +1,253 @@
+
+> Please note the zip file should not exceed 1 GB.
+
+## Enable Local Cache
+
+When this feature is enabled, the site content is read, written from the local virtual machine instance instead of fetching from Azure storage (where site content is stored). This will reduce the number of recycles required for the app. It can be enabled through Azure portal from the "General -> Application settings". On this page under the App settings section add `WEBSITE_LOCAL_CACHE_OPTION` as key and `"Always"` as value. Also add the `WEBSITE_LOCAL_CACHE_SIZEINMB` with a desired local cache size value up to 2000MB (if not provided, it defaults to 300 MB). It helps to provide the cache size specially when the site contents are more than 300 MB. Ensure that site contents are less than 2000MB for this feature to work. Also it is a good practice to keep it as a slot setting so that it does not get removed with a swap.
+**The most important thing to keep in mind here** is that app should not be doing local disk writes for state persistence of its data/transactions.
+External storage like storage containers, db or cosmosDB should be used for storage purposes.
+
+> Please note that the behavior of Local Cache depends on the language and CMS you are using. For best results, we recommend using it for .net and .netcore apps as long as local writes are not being done by the app.  
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/local-cache.png)
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/local-cache-2.png)
+
+**Learn More**
+
+- [Azure App Service Local Cache overview](https://docs.microsoft.com/en-us/azure/app-service/overview-local-cache)
+
+## Auto Heal
+
+Enable auto heal on the app. I would recommend to create an auto heal mitigation rule by going to "Diagnose and Solve problems" section, selecting the "Diagnostic Tools" tile and then "Auto Healing" under Proactive Tools section.
+
+![multiple-instances]({{site.baseurl}}/media/2020/04/autoheal.jpg)
+
+Below are example filter values to set up, however if some other value of error code and frequency suits your application, please modify accordingly:
+
+| Condition        |       Value    |
+|------------------|----------------|
+| Request Count    | 70             |
+| Status Code      | 500            |
+| Sub-status code  | 0              |
+| Win32-status code| 0              |
+| Frequency in seconds| 60          |
+| Request Count    | 70             |
+| Request Count    | 70             |
+| Request Count    | 70             |
+
+Configure Action
+<br>Recycle Process
+<br>Override when Action Executes 
+<br>Startup Time for process before auto heal executes: 3600 seconds (1 hour)
```

Comment:
What is this section explaining? Are these all the actions that they can choose from, or are these the three that the user should enable?

Reply:
This is the recommended action that the customer should take. I fixed the styling to make it more obvious + a mistake that was there.